design-proposals/persist-vmi-firmware-uuid.md

```diff
@@ -0,0 +1,129 @@
+# Overview
+This proposal introduces a mechanism to persist the firmware UUID of a Virtual Machine Instance (VMI) in KubeVirt.
+By storing the firmware UUID, we ensure that it remains consistent across VMI restarts.
+which is crucial for applications and services that rely on the UUID for identification or licensing purposes.
```

Comment:
I wonder if it's worth mentioning cloud-init, as that would be the main issue with the current approach since the firmware UUID directly corresponds to the instance ID: https://github.com/kubevirt/kubevirt/blob/ced9fe1b5104daa431bed6b65ea690639fe3b79f/pkg/cloud-init/cloud-init.go#L117-L122

Reply:
not sure i understand how the cloud-init impact the scope of this proposal
could you please elaborate more on this?

Reply:
Sorry if it wasn't clear. 

You wrote that the firmware UUID is crucial for applications and service that rely on it for identification or licensing purposes. I raised that cloud-init, which is perhaps the most common case of using the firmware UUID for anything, would be a good example.

Reply:
Or even registration to Red Hat Satellite would be a typical application where VMs can be registered and relies on the host UUID.